Content.Shared/GameObjects/Components/Disposal/SharedDisposalUnitComponent.cs

```diff
@@ -57,6 +68,47 @@ public enum PressureState
             Pressurizing
         }
 
+        bool ICollideSpecial.PreventCollide(IPhysBody collided)
+        {
+            if (IsExiting(collided.Entity)) return true;
+            if (!Owner.TryGetComponent(out IContainerManager manager)) return false;
+
+            var containedEntities = manager.GetContainer(Name).ContainedEntities.ToList<IEntity>();
```

Comment:
Either the container as an `IContainer` or its contents should be declared as a property in `SharedDisposalUnitComponent` for shared code to use more easily. This could break if the container name changes, for example.